discovery_engine/lib/src/domain/models/active_data.dart

```diff
@@ -26,6 +26,10 @@ part 'active_data.g.dart';
 /// Additional data pertaining to active documents.
 @HiveType(typeId: activeDocumentDataTypeId)
 class ActiveDocumentData with EquatableMixin {
+  /// S-mBert Embedding
+  ///
+  /// Is a Float32List cast to a Uint8List.
```

Comment:
i'm not sure if i get this right, is the `Uint8List` the internal representation of the embedding in hive? i guess hive can't store the `Float32List` directly?

Reply:
Hive does only have a few basic types so it will always store `Float32List` as `Uint8List`.

But we can make a `Float32List` adapter (there might also be an adapter), I can look into
this later today.

Generally given that it's a local database we probably want to store it in native endian, and
or given that ARM default to little endian in LE.

Reply:
Ok Then we need the embedding type adapter :/

Reply:
In follow up PR?

Reply:
Fine for me!